src/main/java/com/google/summit/ast/declaration/Declaration.kt

```diff
@@ -30,12 +29,36 @@ import com.google.summit.ast.modifier.Modifier
  * @param loc the location in the source file
  */
 sealed class Declaration(val id: Identifier, loc: SourceLocation) : NodeWithSourceLocation(loc) {
```

Comment:
Could the Declaration be Modifiable? (Aren't all of the concrete classes eventually implementing Modifiable?)

Reply:
This is currently the case, but when variable declarations are refactored to match this design, `VariableDeclarationStatement` (or something equivalent) will be the `Modifiable` and `VariableDeclaration` will extend only `Declaration`.

Reply:
It depends on the delegation design we go with. If we implement `Modifiable` directly for delegation (option 1 or 4), it's true that all current `Declaration`s would be `Modifiable` whether or not they contain the `List<Modifier>`. (In option 2 or 3, `FieldDeclaration` would not be a `Modifiable`.)

To me, it seems like `Declaration`s should only be deemed `Modifiable` in particular cases, e.g. `FieldDeclaration`s are because we know they are always contained in `FieldDeclarationGroup`s which are `Modifiable`. Should we assume that all `Declaration`s in general will either be `Modifiable` or be contained by a `Modifiable` node?

Reply:
Given that types that implement `Declaration` are what you see here (assuming that the Apex language isn't likely to evolve much), in this context `Declaration` apparently implies `Modifiable`. But I agree that it is not inherent to the concept of a declaration.

I'm fine with whatever direction you choose. Just wanted to raise the question.